pkg/pipeline/source/sdk/appwriter.go

```diff
@@ -354,38 +385,46 @@ func (w *AppWriter) pushBlankFrame(timestamp uint32) error {
 
 func (w *AppWriter) push(packets []*rtp.Packet, blankFrame bool) error {
 	for _, pkt := range packets {
-		// record timestamp diff
-		if w.rtpStep == 0 && !blankFrame && w.lastRTP != 0 && pkt.SequenceNumber == w.lastSN+1 {
-			w.rtpStep = pkt.Timestamp - w.lastRTP
-		}
-
-		// record SN and TS
-		w.lastSN = pkt.SequenceNumber
-		if w.lastRTP > wrapCheck && pkt.Timestamp < wrapCheck {
-			w.rtpWrap++
-		}
-		w.lastRTP = pkt.Timestamp
-
 		if !blankFrame {
 			// update sequence number
 			pkt.SequenceNumber += w.snOffset
+
+			// record max frame duration
+			if w.lastTS != 0 && pkt.SequenceNumber == w.lastSN+1 {
+				if frameDurationRTP := pkt.Timestamp - w.lastTS; frameDurationRTP > w.frameDurationRTP && frameDurationRTP < w.clockRate/60 {
```

Comment:
Just curious about the `w.clockRate/60` test here?

Reply:
That actually isn't supposed to be there - the goal was to make sure it's reasonable, will change to `< w.clockRate` to keep it < 1 second